src/Stratis.Features.FederatedPeg/Collateral/CollateralChecker.cs

```diff
@@ -281,7 +285,7 @@ public void Dispose()
             this.signals.Unsubscribe(this.memberAddedToken);
             this.signals.Unsubscribe(this.memberKickedToken);
 
-            this.cancellationSource.Cancel();
```

Comment:
@quantumagi why doesn't calling Cancel on the token source work here?

Reply:
`isDisposing` is a more succinct way of connecting the disposal of the object to the termination of the async loop. It does not care whether the application is stopping or not. It's either this or introduce another cancellation token. I opted for a simple boolean.